src/mds/nameserver2/namespace_storage.cpp

```diff
@@ -191,7 +205,27 @@ StoreStatus NameServerStorageImp::RenameFile(const FileInfo &oldFInfo,
                   static_cast<int>(newStoreKey.size()),
                   static_cast<int>(encodeNewFileInfo.size())};
     std::vector<Operation> ops{op1, op2};
-    int errCode = client_->TxnN(ops);
+    errCode = client_->TxnN(ops);
+    }
+    else{
+    // update mysql
+    mysqlclient_->conn_->setAutoCommit(false);
+    try
+    {
+        mysqlclient_->Delete(oldStoreKey);
```

Comment:
`mysqlclient_` can use the same interface as `TxN(ops)` because `ops` is independent of the client type and only packages the operations.